Plugins/SpatialGDK/SpatialGDKEditorToolbar/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp

```diff
@@ -256,19 +256,35 @@ void FSpatialGDKEditorToolbarModule::GenerateInteropCodeButtonClicked()
 	ShowTaskStartNotification("Generating Interop Code");
 	bInteropCodeGenRunning = true;
 
-	AsyncTask(ENamedThreads::GameThread, [this] {
-		bool bSuccess = SpatialGDKGenerateInteropCode();
+	ClassHeaderMap InteropGeneratedClasses;
+	if (!GenerateClassHeaderMap(InteropGeneratedClasses))  // Checks that all classes are found and generate the class mapping.
+	{
+		UE_LOG(LogSpatialGDKInteropCodeGenerator, Error, TEXT("Not all classes found; check your DefaultEditorSpatialGDK.ini file."));
+		ShowFailedNotification("Interop Codegen Failed");
+		return;
+	}
+
+	TFunction<bool()> CodegenTask = [InteropGeneratedClasses]()
```

Comment:
Pass by &?

Reply:
InteropGeneratedClasses could go out of scope before the Async task completes